python/lsst/resources/s3.py

```diff
@@ -192,6 +227,22 @@ def mkdir(self) -> None:
             self.client.put_object(Bucket=self.netloc, Key=self.relativeToPathRoot)
 
     @backoff.on_exception(backoff.expo, all_retryable_errors, max_time=max_retry_time)
+    def _download_file(self, local_file: ResourcePath, progress: Optional[ProgressPercentage]) -> None:
+        """Download the remote resource to a local file.
+
+        Helper routine for _as_local to allow backoff without regenerating
+        the temporary file.
+        """
+        try:
+            self.client.download_fileobj(self.netloc, self.relativeToPathRoot, local_file, Callback=progress)
+        except (
+            self.client.exceptions.NoSuchKey,
+            self.client.exceptions.NoSuchBucket,
+        ) as err:
+            raise FileNotFoundError(f"No such resource: {self}") from err
+        except ClientError as err:
+            _translate_client_error(err, FileNotFoundError(f"No such resource: {self}"))
```

Comment:
At a minimum, maybe the message should be "Resource inaccessible"?

Reply:
I now simply re-raise.